builder/vmware/common/step_vnc_connect.go

```diff
@@ -0,0 +1,137 @@
+package common
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/url"
+
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+	"github.com/mitchellh/go-vnc"
+	"golang.org/x/net/websocket"
+)
+
+type StepVNCConnect struct {
+	VNCEnabled       bool
+	VNCOverWebsocket bool
+	DriverConfig     *DriverConfig
+}
+
+func (s *StepVNCConnect) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	if !s.VNCOverWebsocket && !s.VNCEnabled {
+		return multistep.ActionContinue
+	}
+	ui := state.Get("ui").(packer.Ui)
+
+	var c *vnc.ClientConn
+	var err error
+
+	if s.VNCOverWebsocket {
```

Comment:
More a general question, is there a deprecation plan for removing VNC? I'll have to check for sure but I'm guessing VNC over websocket is supported in all non-EOL ESXi (6.5+).

Reply:
We haven't thought about deprecating VNC yet, but we can discuss that. This depends on the version of hosts we want Packer to support building VMs to. If we come to decide that we will officially support ESXi 6.5+ I don't see why keep the VNC. Do you have any specific concerns about this?

Reply:
Main concern is maintaining the 3 paths (VNC, USB keyboard, VNC over WS) as your changes has to check/override depending on the configuration. Just seems like technical debt into the future. Could add a comment about VNC being deprecated in the future to give people time to switch their configs to the newer mechanisms.

Reply:
Yes, you're right. I might decide if we want to really keep the USB Keyboard here. I added it as an option for 6.7+ hosts that don't have VNC, but with this new vnc_over_websocket option that won't be necessary anymore, and like you said it can make configs confusing.